client/src/pages/Write.jsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import ReactQuill from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import { useLocation, useNavigate } from "react-router-dom";
+
+const Write = () => {
+  const { state } = useLocation();
+
+  const navigate = useNavigate();
+
+  const [title, setTitle] = useState(state?.title || "");
+  const [value, setValue] = useState(state?.content || "");
+  const [file, setFile] = useState(null);
+
+  const uploadImage = async () => {
+    try {
+      const formData = new FormData();
+      formData.append("file", file);
+      const response = await fetch("http://localhost:8080/api/v1/upload", {
+        method: "POST",
+        body: formData,
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        return data;
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const imgUrl = await uploadImage();
+    try {
+      let response;
+      state
+        ? (response = await fetch(
+            `http://localhost:8080/api/v1/posts/${state.id}`,
+            {
+              method: "PUT",
+              headers: {
+                "Content-type": "application/JSON",
+              },
+              body: JSON.stringify({
+                title,
+                content: value,
+                img: file ? imgUrl : "",
+              }),
+            }
+          ))
+        : (response = await fetch("http://localhost:8080/api/v1/posts", {
+            method: "POST",
+            headers: {
+              "Content-type": "application/JSON",
+            },
+            body: JSON.stringify({
+              title,
+              content: value,
+              img: file ? imgUrl : "",
+            }),
+          }));
```

Comment:
The only differences in these `fetch()` calls are the `method` and URLs. I'd write an `if` statement that sets variables, and use the variables in your `fetch()` arguments.